nixos/modules/services/networking/fastnetmon-advanced.nix

```diff
@@ -0,0 +1,248 @@
+    mkdir hostgroup
+    ${lib.concatImapStringsSep "\n" (pos: hostgroup: ''
+      jq -s add ${default_configs}/hostgroup/0.json ${pkgs.writeText (baseNameOf "hostgroup-${toString (pos - 1)}-add.json") (builtins.toJSON hostgroup)} > hostgroup/${toString (pos - 1)}.json
+    '') hostgroups}
+    mkdir bgp
+    ${lib.concatImapStringsSep "\n" (pos: bgp: ''
+      jq -s add ${default_configs}/bgp/0.json ${pkgs.writeText (baseNameOf "bgp-${toString (pos - 1)}-add.json") (builtins.toJSON bgp)} > bgp/${toString (pos - 1)}.json
+    '') bgpPeers}
+    tar -cf $out main.json ${lib.concatImapStringsSep " " (pos: _: "hostgroup/${toString (pos - 1)}.json") hostgroups} ${lib.concatImapStringsSep " " (pos: _: "bgp/${toString (pos - 1)}.json") bgpPeers}
+  '';
+
+  hostgroups = lib.mapAttrsToList (name: hostgroup: { inherit name; } // hostgroup) cfg.hostgroups;
+  bgpPeers = lib.mapAttrsToList (name: bgpPeer: { inherit name; } // bgpPeer) cfg.bgpPeers;
+
+in {
+  options.services.fastnetmon-advanced = with lib; {
+    enable = mkEnableOption (mdDoc "Enable the fastnetmon-advanced DDoS Protection daemon");
+
+    settings = mkOption {
+      description = mdDoc "Extra configuration options to declaratively load into FastNetMon Advanced";
+      type = settingsFormat.type;
+      default = {};
+    };
+    hostgroups = mkOption {
+      description = mdDoc "Hostgroups to declaratively load into FastNetMon Advanced";
+      type = types.attrsOf settingsFormat.type;
+      default = {};
+    };
+    bgpPeers = mkOption {
+      description = mdDoc "BGP Peers to declaratively load into FastNetMon Advanced";
+      type = types.attrsOf settingsFormat.type;
+      default = {};
+    };
+
+    enableAdvancedTrafficPersistence = mkOption {
+      description = mdDoc "Store historical flow data in clickhouse";
+      type = types.bool;
+      default = false;
+    };
+  };
+
+  config = lib.mkMerge [ (lib.mkIf cfg.enable {
+    services.fastnetmon-advanced.settings.pid_path = "/run/fastnetmon/fastnetmon.pid";
+
+    environment.systemPackages = with pkgs; [
+      fastnetmon-advanced
+    ];
+
+    # FastNetMon Advanced *requires* a MongoDB to store its configuration in
+    # We use PostgreSQL + FerretDB instead of MongoDB because MongoDB is not in the binary cache: SSPL license is not OSI approved thus considered unfree
+    services.postgresql = {
+      enable = true;
+      ensureDatabases = [ "fastnetmon" ];
+      ensureUsers = [
+        {
+          name = "fastnetmon";
+          ensurePermissions = { "DATABASE fastnetmon" = "ALL PRIVILEGES"; };
+        }
+      ];
+    };
+
+    systemd.services.ferretdb = {
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.ferretdb}/bin/ferretdb";
+        DyanmicUser = true;
+      };
+      environment = {
+        FERRETDB_POSTGRESQL_URL = "postgresql://localhost:5432";
+        FERRETDB_LOG_LEVEL = "debug";
+        FERRETDB_TELEMETRY = "disable";
+      };
+    };
```

Comment:
This should probably go in another module

Reply:
I created a ferretdb module here : https://github.com/NixOS/nixpkgs/pull/231138
Would be interested to have your feedback and know if it would fit your usecase